src/components/create-study-form.js

```diff
@@ -185,7 +198,42 @@ export const CreateStudyForm = () => {
     };
 
     const handleStudyNameChanges = (e) => {
-        setStudyName(e.target.value);
+        const name = e.target.value;
+        setStudyName(name);
+
+        setStudyNameChecked(false);
+        setLoadingCheckStudyName(true);
+
+        clearTimeout(timer.current);
+        timer.current = setTimeout(() => {
+            updateStudyFormState(name);
+        }, 700);
+    };
```

Comment:
setTimeout just waits the 700ms and executes the callback once. It's not like setInterval. Since the timer is cancelled at every keystroke, this means it checks the name when the user paused typing